http_handlers.go

```diff
@@ -49,24 +60,44 @@ func webhook(c *gin.Context) {
 			// get chart url
 			generatorUrl, err := url.Parse(alert.GeneratorURL)
 			if err != nil {
-				err = errors.Wrap(err, "Could not get generator url")
-				_ = bugsnag.Notify(err)
+				err = errors.Wrap(err, "Could not parse generator url")
+				_ = bugsnag.Notify(err, ctx,
+					bugsnag.MetaData{
+						"Alert": {
+							"Name": alertName,
+							"GeneratorURL": alert.GeneratorURL,
+						},
+					})
 				clog.Error(err.Error())
 			}
 
 			// from the chart url get the expressions to build the charts
 			generatorQuery, err := url.ParseQuery(generatorUrl.RawQuery)
 			if err != nil {
-				err = errors.Wrap(err, "Could not get query from generator url")
-				_ = bugsnag.Notify(err)
+				err = errors.Wrap(err, "Could not parse query from generator url")
+				_ = bugsnag.Notify(err, ctx,
+					bugsnag.MetaData{
+						"Alert": {
+							"Name": alertName,
+							"RawQuery": generatorUrl.RawQuery,
+						},
+					})
 				clog.Error(err.Error())
 			}
 
 			// shorten generator URL
-			err, n := cli.ReplaceLinks(c, alert.GeneratorURL)
+			err, n := cli.ReplaceLinks(ctx, alert.GeneratorURL)
```

Comment:
I haven't read up on it but are we sure we want to be passing in this context here?

Reply:
`cli.Submit` requires a context which is called in `cli.ReplaceLinks`

Reply:
Yeah, but without having looked into it I'd have assumed that most of the time you'd just use the overall gin context?

Reply:
(I've never used gin, so this change could indeed be perfectly correct. It just seemed like an odd thing for the framework to neccesitate on the face of it.)

Reply:
ah right yeh, this https://stackoverflow.com/a/69878290 seems to suggest to use the request context so things are terminated correctly. I guess i just switched everything to use the request context because bugsnag wanted to use that

Reply:
looks like the gin context is for passing variables round https://github.com/gin-gonic/gin/blob/master/context.go#L46 and the request context is a normal context.Context type

Reply:
im not sure which is better but the latter seems more correct to me